src/Artifacts.jl

```diff
@@ -314,16 +315,27 @@ function download_artifact(
         msg  = "Tree Hash Mismatch!\n"
         msg *= "  Expected git-tree-sha1:   $(bytes2hex(tree_hash.bytes))\n"
         msg *= "  Calculated git-tree-sha1: $(bytes2hex(calc_hash.bytes))"
-        # Since tree hash calculation is still broken on some systems, e.g. Pkg.jl#1860,
-        # and Pkg.jl#2317, we allow setting JULIA_PKG_IGNORE_HASHES=1 to ignore the
-        # error and move the artifact to the expected location and return true
-        ignore_hash = Base.get_bool_env("JULIA_PKG_IGNORE_HASHES", false)
+        # actual and expected artifiact paths
+        src = artifact_path(calc_hash; honor_overrides=false)
+        dst = artifact_path(tree_hash; honor_overrides=false)
+        # Since tree hash calculation is rather fragile and file system dependent,
+        # we allow setting JULIA_PKG_IGNORE_HASHES=1 to ignore the error and move
+        # the artifact to the expected location and return true
+        ignore_hash = if get(ENV, "JULIA_PKG_IGNORE_HASHES", "") == ""
+            # default: false except Windows users who can't symlink
+            Sys.iswindows() && !mktempdir(can_symlink, dirname(src))
+        else
+            Base.get_bool_env("JULIA_PKG_IGNORE_HASHES", ignore_hash)
+            ignore_hash === nothing && @error(
+                "Invalid ENV[\"JULIA_PKG_IGNORE_HASHES\"] value",
+                ENV["JULIA_PKG_IGNORE_HASHES"],
+            )
+            ignore_hash = something(ignore_hash, false)
+        end
```

Comment:
I think this does what you want?
```suggestion
	default = Sys.iswindows() && !mktempdir(can_symlink, dirname(src))
	ignore_hash = Base.get_bool_env("JULIA_PKG_IGNORE_HASHES", default)
```

Reply:
That's what I had at first, but there are two issues:

1. I don't want to even try to make a symlink if the variable is set.
2. If the variable is set but has an invalid value, I don't want this code to inscrutably error.

Reply:
Sounds reasonable https://github.com/JuliaLang/julia/pull/52950